openvino/itt/CMakeLists.txt

```diff
@@ -21,9 +21,34 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 file(GLOB_RECURSE SOURCES "src/*.cpp")
 
 if(ENABLE_PROFILING_ITT)
-    find_package(ITT)
-    if(NOT ITT_FOUND)
-        message(WARNING "Profiling option enabled, but no ITT library was found")
+    if(DEFINED INTEL_VTUNE_DIR OR DEFINED ENV{INTEL_VTUNE_DIR})
+        find_package(ITT)
+        if(NOT ITT_FOUND)
+            message(WARNING "Profiling option enabled, but no ITT library was found under INTEL_VTUNE_DIR")
+        endif()
+    else()
+        include(ExternalProject)
+        set(ITTAPI_BINARY_DIR  ${CMAKE_CURRENT_BINARY_DIR}/ittapi/build)
+        set(ITTNOTIFY_LIBRARY  ${ITTAPI_BINARY_DIR}/bin/${CMAKE_FIND_LIBRARY_PREFIXES}ittnotify${CMAKE_STATIC_LIBRARY_SUFFIX})
+        ExternalProject_Add(
+            ittapi_external
+            GIT_REPOSITORY https://github.com/intel/ittapi.git
+            GIT_TAG v3.18.6
+            LOG_DOWNLOAD ON
+            LOG_CONFIGURE ON
+            LOG_BUILD ON
+            INSTALL_COMMAND ""
+            UPDATE_COMMAND ""
+            CMAKE_GENERATOR ${CMAKE_GENERATOR}
+            CMAKE_GENERATOR_PLATFORM ${CMAKE_GENERATOR_PLATFORM}
+            CMAKE_GENERATOR_TOOLSET ${CMAKE_GENERATOR_TOOLSET}
+            BINARY_DIR ${ITTAPI_BINARY_DIR}
+            EXCLUDE_FROM_ALL TRUE
+            BUILD_BYPRODUCTS ${ITTNOTIFY_LIBRARY})
+        add_library(ittnotify INTERFACE)
+        add_dependencies(ittnotify ittapi_external)
+        target_link_libraries(ittnotify INTERFACE ${ITTNOTIFY_LIBRARY})
```

Comment:
maybe replace `INTERFACE` library with properly imported `IMPRORTED` target? Combination of interface + manual link with static library does not look native to cmake.